### DISCUSSION TITLE
How to add SaveChanges interceptors when using AddSqlServerDbContext

### DISCUSSION THREAD

**Discussion by jstynder** [microsoft/aspire#4202] [Q&A]
When adding interceptors this would be the way I do It now:```

  services.AddDbContextPool<T>((serviceProvider, optionsBuilder) =>
      optionsBuilder
          .UseSqlServer(connectionString)
          .AddInterceptors(serviceProvider.GetRequiredService<SecondLevelCacheInterceptor>()));
```
How can I access the serviceProvider when using AddSqlServerDbContext

**Comment by davidfowl** [comment]
See https://github.com/dotnet/docs-aspire/issues/1601#issuecomment-2333380717
